### ISSUE TITLE
Generate RBAC based on the current config

### ISSUE THREAD

**Issue by tfadeyi** [jetstack/jetstack-secure#250]
Generate RBAC based on the current config, and in such a way that the code can be shared with the backend.

**Comment by j-fuentes** [comment]
More context on this one.

At

**Comment by charlieegan3** [comment]
I've been working a little with Wenlin on this today and my plan was, if there was more time, to build this incrementally with TDD in stages, e.g.

1. GVRs in, RBAC string templated out
2. GVRs in, RBAC structs out
3. GVRs and Namespaces in, scoped rbac generated based on some list of known namespaces in the case of excluded namespaces
4. means of getting GVRs from loaded dynamic data gatherers.
5. adding a command to spit this all out.
6.

**Comment by charlieegan3** [comment]
You are correct, I was imagining that it'd be possible to supply a list of namespaces that comes from kubectl get ns if the agent were running locally using the installers credentials.

I'm thinking that the first use of this is going to be a command in the agent, and if the agent, running in that command had access to the installers creds, it could get a list of all namespaces.

We might not be able to do this when using it from the platform side, however I think it's a feature worth adding here in the agent.
